scripts/tools/goal_obstacle.py

```diff
@@ -0,0 +1,175 @@
+        self.random_movement = False
+        self.robot_index = 0
+        self.obstacle_index = 0
+        self.last_time = time.time()
+        self.inc_vel = 0.43
+
+    def process(self):
+
+        # keys reading part
+        chars = self.raw_key.get_num_chars()
+        if len(chars) > 0:
+            # select obstacle
+            if chars == Keys.q:
+                self.obstacle_index += 1
+                self.obstacle_index = self.obstacle_index % self.obstacle_cant
+            # select robot goal
+            if chars == Keys.o:
+                self.robot_index += 1
+                self.robot_index = self.robot_index % self.robot_cant
+            # control goal
+            if chars == Keys.UP_ARROW:
+                self.goal_vel_list[self.robot_index][1] += self.inc_vel
+            if chars == Keys.DOWN_ARROW:
+                self.goal_vel_list[self.robot_index][1] -= self.inc_vel
+            if chars == Keys.RIGHT_ARROW:
+                self.goal_vel_list[self.robot_index][0] += self.inc_vel
+            if chars == Keys.LEFT_ARROW:
+                self.goal_vel_list[self.robot_index][0] -= self.inc_vel
+            # control obstacle
+            if chars == Keys.w:
+                self.obstacle_vel_list[self.obstacle_index][1] += self.inc_vel
+            if chars == Keys.s:
+                self.obstacle_vel_list[self.obstacle_index][1] -= self.inc_vel
+            if chars == Keys.a:
+                self.obstacle_vel_list[self.obstacle_index][0] -= self.inc_vel
+            if chars == Keys.d:
+                self.obstacle_vel_list[self.obstacle_index][0] += self.inc_vel
+            # positions and movement options
+            if chars == Keys.r:
+                # give every obstacle a random position
+                for i in range(self.obstacle_cant):
+                    self.obstacle_vel_list[i] = array([0., 0., 0., 0., 0., 0.])
+                    self.obstacle_pose_list[i][:2, 3] = 10 * random(2) - array(
+                        [5, 5])
+            if chars == Keys.f:
+                # give every obstacle a random movement behaviour
+                self.random_movement = not self.random_movement
+                for i in range(self.obstacle_cant):
+                    self.obstacle_vel_list[i][:2] = 0. * random(2) - array(
+                        [0.025, 0.025])
+
+        # time meassurement
+        self.cur_time = time.time()
+        dt = self.cur_time - self.last_time
+        self.last_time = self.cur_time
+
+        # make obstacles move randomly if necesary
+        if self.random_movement:
+            for i in range(self.obstacle_cant):
+                self.obstacle_vel_list[i][:2] -= 0.1 * self.obstacle_vel_list[
+                    i][:2]
+                self.obstacle_vel_list[i][:2] += 0.6 * random(2) - array(
+                    [0.3, 0.3])
+                if norm(self.obstacle_pose_list[i][:2, 3]) > 12:
+                    self.obstacle_vel_list[i][:2] += old_div(
```

Comment:
old_div not required, please use the regular division operator `/`

Reply:
Good catch. Now all files are PEP238 compliant.